server/punishments.ts

```diff
@@ -679,6 +679,7 @@ export const Punishments = new class {
 		const ipStr = typeof user !== 'string' ? ` [${(user as User).latestIp}]` : '';
 		const roomid = typeof room !== 'string' ? (room as Room).roomid : room;
 		Rooms.global.modlog(`(${roomid}) AUTO${namelock ? `NAME` : ''}LOCK: [${userid}]${ipStr}: ${reason}`);
+		Chat.parse('/savereplay', room as Room, user as User, (user as User).connections[0]);
```

Comment:
this cast isn't always right, see: https://github.com/Zarel/Pokemon-Showdown/blob/c990e015dbff2cece541833e77cb4a903a61cd2a/server/punishments.ts#L1452

Reply:
Oh that's fair.